modello-core/src/main/java/org/codehaus/modello/plugin/CachingWriter.java

```diff
@@ -25,31 +48,20 @@ public CachingWriter( BuildContext buildContext, Path path, Charset charset )
     @Override
     public void close() throws IOException
     {
-        String str = getBuffer().toString();
-        if ( Files.exists( path ) )
+        byte[] data = getBuffer().toString().getBytes( charset );
+        if ( Files.exists( path ) && Files.size( path ) == data.length )
         {
-            String old = readString( path, charset );
-            if ( str.equals( old ) )
+            byte[] ba = Files.readAllBytes( path );
+            if ( Arrays.equals( data, ba ) )
             {
                 return;
             }
         }
-        writeString( path, str, charset );
+        Files.write( path, data );
         if ( buildContext != null )
```

Comment:
It would be nice to have debug or something logging so the users can recognize why the file was overwtitten or not

Reply:
You mean log the diff ? I'm not sure that's really reasonable.

Reply:
I mean something like "the contents of the file... matches, skipping the overwrite" or "the contents differs => overwriting the file".
I do not suggest printing the diff or something.